flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeDuplicator.java

```diff
@@ -107,22 +109,32 @@ public LogicalType visit(StructuredType structuredType) {
 					a.getType().accept(this));
 			})
 			.collect(Collectors.toList());
-		final StructuredType.Builder builder = new StructuredType.Builder(
-			structuredType.getObjectIdentifier(),
-			attributes);
-		builder.setNullable(structuredType.isNullable());
-		builder.setFinal(structuredType.isFinal());
-		builder.setInstantiable(structuredType.isInstantiable());
-		builder.setComparision(structuredType.getComparision());
+		final Optional<ObjectIdentifier> identifier = structuredType.getOptionalObjectIdentifier();
+		final Optional<Class<?>> implementationClass = structuredType.getImplementationClass();
+
+		final StructuredType.Builder builder;
```

Comment:
nit: This is really nitpicking. How about extracting the instantiation of new `builder` and duplicating the attributes to separate methods. The visit method is already quite long.